libs/src/utils/network.js

```diff
@@ -62,9 +62,18 @@ async function timeRequestsAndSortByVersion (requests, timeout = null) {
     // If health check failed, send to back of timings
     if (!a.response) return 1
     if (!b.response) return -1
+
     // Sort by highest version
     if (semver.gt(a.response.data.data.version, b.response.data.data.version)) return -1
     if (semver.lt(a.response.data.data.version, b.response.data.data.version)) return 1
+
+    // Sort by content node transcode queue load
+    // defined as the ratio of (active + waiting transcodes) / (transcode slots, same as number of cpu cores)
+    const healthRatioA = ((a.response.data.data.transcodeActive + a.response.data.data.transcodeWaiting) / a.response.data.data.numberOfCPUs)
```

Comment:
wait this method is shared isn't it? should we be checking on whether these fields exist?

Reply:
also for backwards compatibility, this would throw if these fields don't exist right? we can't release this until all cnodes update. i think the libs PR should be separated from the cnode PR in that case

Reply:
if one/all of these fields don't exist or are 0, as far as the sort is concerned it does nothing so it should  pass through to the time tiebreaker

also it doesn't error if these fields don't exist since pulling a nonexistent property off an object is undefined. it's only when you pull a property off another property that doesn't exist it errors. but i can add validation to make it cleaner

Reply:
const healthRatioA = ((a.response.data.data.transcodeActive + a.response.data.data.transcodeWaiting) / a.response.data.data.numberOfCPUs)

this would be undefined / some value

won't that throw?

Reply:
it actually doesn't, it becomes NaN, but let me just add some checking. i'm also happy to separate but I don't think we need to. i think just defaulting the ratio to 1 to simulate all transcode slots taken should be sufficient.

in the case that a node upgrades, we'll get the real data and since it's likely to be lower than 1 it'll be selected.
in the case everyone upgrades, we'll have accurate data
in the case nobody upgrades, it'll be the same for everyone and the tie breaker will be time

Reply:
default sounds reasaonble